packages/hash/shared/src/config.ts

```diff
@@ -0,0 +1,6 @@
+export const frontendDomain = process.env.FRONTEND_DOMAIN ?? "localhost:3000";
```

Comment:
As mentioned in the other comment thread about the URIs, this ENV var would not be exposed when nextjs compiles the frontend as it's not prefixed with `NEXT_PUBLIC_`. This may be a problem when hosting with another URI, and I think we should make that clear somehow as this is a shared export between front/backend.

Reply:
Perhaps this would do?
```suggestion
export const frontendDomain = process.env.FRONTEND_DOMAIN ?? process.env.NEXT_PUBLIC_FRONTEND_URL ?? "localhost:3000";
```

Reply:
If your suggestion is to replace the frontend's `FRONTEND_URL` definition with this, wouldn't that mean we have to host a dedicated backend service for each vercel staging environment? (i.e. for every open PR)

Reply:
I guess it would be the case since the backend checks URIs against a regex to determine if it's a "valid type uri" and therefore would reject anything that's `hash-*.stage.hash.ai/` unless we make the backend regex really permissive (which we could for a staging environment)

We would have to create an env var for `FRONTEND_DOMAIN` in vercel if we want to use a shared staging environment right now, though. And it would have to match the regex of whatever staging backend we stand up to support frontends

Reply:
Ah I misunderstood your suggestion, thanks for spotting. Addressed in https://github.com/hashintel/hash/pull/1193/commits/e5e25825a21af3e7b4e7a8eb1a20298721e179de